server/src/testE2E/java/com/flipkart/varadhi/SubscriptionTests.java

```diff
@@ -0,0 +1,219 @@
+package com.flipkart.varadhi;
+
+import com.flipkart.varadhi.entities.*;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import javax.ws.rs.core.GenericType;
+import javax.ws.rs.core.Response;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+
+import static com.flipkart.varadhi.entities.VersionedEntity.INITIAL_VERSION;
+import static com.flipkart.varadhi.entities.VersionedEntity.NAME_SEPARATOR_REGEX;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class SubscriptionTests extends E2EBase {
+
+    private static final Endpoint endpoint;
+    private static Org o1;
+    private static Team o1t1;
+    private static Project o1t1p1;
+    private static TopicResource p1t1;
+    private static TopicResource p1t2;
+
+    static {
+        try {
+            endpoint = new Endpoint.HttpEndpoint(new URL("http", "localhost", "hello"), "GET", "", 500, 500, false);
+        } catch (MalformedURLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private final RetryPolicy retryPolicy = new RetryPolicy(
+            new CodeRange[]{new CodeRange(500, 502)},
+            RetryPolicy.BackoffType.LINEAR,
+            1, 1, 1, 1
+    );
+    private final ConsumptionPolicy consumptionPolicy = new ConsumptionPolicy(1, 1, false, 1, null);
+
+    @BeforeAll
+    public static void setup() {
+        o1 = new Org("public", 0);
+        o1t1 = new Team("team1", 0, o1.getName());
+        o1t1p1 = new Project("default", 0, "", o1t1.getName(), o1t1.getOrg());
+        p1t1 = new TopicResource("topic1", 0, o1t1p1.getName(), false, null);
+        p1t2 = new TopicResource("topic2", 0, o1t1p1.getName(), true, null);
+        makeCreateRequest(getOrgsUri(), o1, 200);
+        makeCreateRequest(getTeamsUri(o1t1.getOrg()), o1t1, 200);
+        makeCreateRequest(getProjectCreateUri(), o1t1p1, 200);
+        makeCreateRequest(getTopicsUri(o1t1p1), p1t1, 200);
+        makeCreateRequest(getTopicsUri(o1t1p1), p1t2, 200);
+    }
+
+    @AfterAll
+    public static void tearDown() {
+        cleanAllSubscriptions(o1t1p1);
+        cleanupTopic(p1t1.getName(), o1t1p1);
+        cleanupTopic(p1t2.getName(), o1t1p1);
+        cleanupProject(o1t1p1);
+        cleanupTeam(o1t1);
+        cleanupOrgs(List.of(o1));
+    }
+
+    private static void cleanAllSubscriptions(Project project) {
```

Comment:
below methods seems candidate for E2EBase.
CleanupOrgs() internally cleans up all entities by walking the hierarchy, so once above is done tearDown() can only call cleanupOrgs().